torch/csrc/jit/script/tree.h

```diff
@@ -56,14 +57,18 @@ struct Tree : std::enable_shared_from_this<Tree> {
   }
   virtual TreeRef map(const std::function<TreeRef(TreeRef)>& fn) {
     (void)fn;
-    return shared_from_this();
+    c10::raw::intrusive_ptr::incref(this); // we are creating a new pointer
+                                           // from a raw `this` pointer
+                                           // so we need to bump the refcount
+                                           // to account for this ownership
+    return TreeRef::reclaim(this);
   }
   template <typename... Args>
-  void match(int k, Args&... args) {
+  void match(int k, Args&... args) const {
```

Comment:
why are these made const in this PR?

Reply:
Honestly, because the compiler refused to accept it without it.  I haven't tracked down what  specific difference between `intrusive_ptr` and `shared_ptr` causes it.